Reflections/Week_4_reflection.md

```diff
@@ -0,0 +1,15 @@
+## Accomplishments
+
+Ultimately this week was a week of decision paralysis. The roadblock that I had faced with implementing the iterator syntax was enough to completely freeze me from getting any meaningful work done. I am also writing this reflection late, as a result of this mental block.
+
+## Challenges
+
+This week had no real challenges in the technical sense, but instead some mental challenges of trying to motivate myself on a problem that had taxed me mentally to work on.
+
+## Next Steps
+
+As of writing this, I have begun working on a file storage interface, and after a small amount of hurdles to jump I was able to continue working on it cleanly without issue. my motivation has improved a lot.
```

Comment:
I think this is a fine next step; please strive for 3 SMART goals in this section in the future!